lib/improver/tests/argparser/test_ArgParser.py

```diff
@@ -0,0 +1,328 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for argparser.ArgParser."""
+
+
+import os
+import unittest
+from mock import patch
+
+from improver.argparser import ArgParser
+
+
+# We might one day want to move this up to a more central place.
+class QuietTestCase(unittest.TestCase):
+
+    """A subclass of unittest.TestCase which prevents writing to stderr and
+    calling of sys.exit."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Patch the class by redirecting stderr to /dev/null, and disabling
+        calls to sys.exit."""
+        cls.stderr_patch = patch('sys.stderr', open(os.devnull, 'w'))
+        cls.exit_patch = patch('sys.exit')
+        cls.stderr_patch.start()
+        cls.exit_patch.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Stop the patches which redirect stderr to /dev/null and prevents
+        sys.exit from being called."""
+        cls.stderr_patch.stop()
+        cls.exit_patch.stop()
+
+
+class Test_init(QuietTestCase):
+
+    """Test the __init__ method."""
+
+    def test_create_argparser_with_no_arguments(self):
+        """Test that creating an ArgParser with no arguments has no
+        arguments."""
+
+        compulsory_arguments = {}
+
+        # it doesn't matter what the centralized arguments are, because we
+        # select None of them
```

Comment:
Not sure this comment is helpful.  You might want something like `Replace class COMPULSORY_ARGUMENTS with mocked empty list to test behaviour`, either here or in the docstring.